### PR TITLE
[image-to-text pipeline] Add conditional text support + GIT

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

The `ImageToText` pipeline can generate text given an image, but oftentimes one wants to make the model continue text given a prompt (like "a photo of"). This PR adds support for conditional text generation given an image.

It also adds support for GIT.

This PR fixes a part of #21110 and is based on #22423.

To do:

- [x] add support for Pix2Struct once design is approved

cc @younesbelkada

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for adding this! Let's wait for @Narsil before merging this

File: src/transformers/pipelines/image_to_text.py
Comment (Narsil) [inline src/transformers/pipelines/image_to_text.py:123]: This is odd.

why `add_special_tokens=False` and then readd them right aftewards  ?
This could become an issue on finetuned model that might use different prompt tokens directly from the configuration.

File: tests/pipelines/test_pipelines_image_to_text.py
Comment (Narsil) [inline tests/pipelines/test_pipelines_image_to_text.py:150]: Can we add `pipeline(..., prompt="some prompt")` in a fast test ? So we make sure we don't break it ?

File: src/transformers/pipelines/image_to_text.py
Comment (Narsil) [inline src/transformers/pipelines/image_to_text.py:131]: Why do we need to pop those ? What happens if we leave them ?

**Review by Narsil — COMMENTED** [review]
Super nice ! Thanks for this ! 

Overall everything is quite clean, I'm just a bit unsure about the tokenization code which seems a bit specific to certain tokenization.

It's ok if we have to do it that way, but maybe we should add a comment with the code because currently it's impossible to know why we don't just do "tokenizer(prompt)".

File: src/transformers/pipelines/image_to_text.py
Comment (NielsRogge) [reply src/transformers/pipelines/image_to_text.py:131]: This was taken from the PR here: https://github.com/huggingface/transformers/pull/22423.

However, it turns out this is very BLIP-specific. The reason is that BLIP uses `BertTokenizer`, which returns `token_type_ids` by default, but `BlipForConditionalGeneration` doesn't accept `token_type_ids`. 

I think the fix here is to include `model_input_names` in the tokenizer_config.json file of the checkpoints on the hub as was done for GIT [here](https://huggingface.co/microsoft/git-base/blob/main/tokenizer_config.json#L5-L8), and eventually remove [this line](https://github.com/huggingface/transformers/blob/c2393cad085e3875ee2206d917d46d15e50602a3/src/transformers/models/blip/processing_blip.py#L44).

@younesbelkada could you take this up?

Comment (Narsil) [reply src/transformers/pipelines/image_to_text.py:131]: Thanks for the explanation, we can leave the clean up for a follow up PR.

Comment (younesbelkada) [reply src/transformers/pipelines/image_to_text.py:131]: Sure @NielsRogge , just merged the PRs on the Hub

**Comment by NielsRogge** [comment]
@Narsil thanks for your review, feel free to approve.

**Review by Narsil — APPROVED** [review]

File: src/transformers/pipelines/image_to_text.py
Comment (sgugger) [inline src/transformers/pipelines/image_to_text.py:136]: `prompt` is not None is tested in every branch here. Can we do a bit `if prompt is not None` then the cascade of tests (which indicates all those models should probably not be in the same Auto class by the way, since they all have different APIs) and then the git part in the else branch? It would be more readable this way.

**Review by sgugger — COMMENTED** [review]
Not sure why I'm specifically asked for review when Amy is on watch, please respect the [calendar](https://www.notion.so/huggingface2/9973fca00bfd43a986daaac9754eb42a?v=cbdda5b50aea441cb0c903f90b7aeca6) when pinging core maintainers unless there is a reason to ping one in particular.

**Comment by NielsRogge** [comment]
Apologies, will take this into account.

**Review by sgugger — APPROVED** [review]

**Pull request merged**
